alf/optimizers/optimizers.py

```diff
@@ -380,6 +428,10 @@ def add_param_group(self, param_group):
         else:
             super(NewCls, self).add_param_group(param_group)
 
+        capacity_ratio = self._capacity_ratio
+        states = self._generate_capacity_mask(capacity_ratio)
+        self._apply_capacity_mask(capacity_ratio, states)
```

Comment:
why are these additional self._apply_capacity_mask needed for parvi_step?

Reply:
Not sure if I understand your question correctly. This is in ``add_param_group``, not in ``parvi_step``. 
The reason to ``_apply_capacity_mask`` in ``add_param_group`` is to correctly handle the parameter values before any forward step.